Makefile

```diff
@@ -1,5 +1,6 @@
 UNAME_S := $(shell uname -s)
 build: 
+	rice embed-go
```

Comment:
Does this indiscriminately bundle everything in the repo? Can we target only a single directory instead?
Where is the bundled data it generates? Doesn't that need to be checked in?

Since we don't need a filesystem interface, is this tool overkill in the first place? I'm concerned about the extra complexity. What if we just had a `map[string]string` with contract names and JSON values?

Reply:
I'm with Jordan, seems a bit unnecessary to add this dependency just to load a file.

Reply:
go rice scans all the go sources for FindBox methods (like this one https://github.com/gochain-io/web3/blob/bundle-abi/abi_wrapper.go#L13) and add folders (only abi folder) and generates rice-box.go file which is just contains stringified content of the files in abi folder. In theory I could replicate the same functionality and use go generate to generate go files with content from json files, but this would replicating existing go-rice, except ofc it would be lighter

Reply:
I don't think we even need go generate. The file contents can just be in a string instead. We're parsing the JSON anyways.

If we wanted to have the parsed abi data already available as native types, then go generate would be useful.

Reply:
Conversely, if we do keep `rice`, we could call it via `go:generate` and commit the result so as not to complicate the build process.